test/e2e/testutils/tester/tester.go

```diff
@@ -16,105 +16,143 @@ package tester
+	ControllerClient client.Client
+	ClusterID        string
+	KubeconfigPath   string
 }
 
+// Environment variable.
+const (
+	namespaceEnvVar      = "NAMESPACE"
+	ClusterNumberVarKey  = "CLUSTER_NUMBER"
+	kubeconfigBaseName   = "liqo_kubeconf"
+	KubeconfigDirVarName = "KUBECONFIGDIR"
+)
+
 var (
 	tester *Tester
 )
 
 // GetTester returns a Tester instance.
-func GetTester(ctx context.Context) *Tester {
+func GetTester(ctx context.Context, controllerClientsPresence, ignoreClusterIDError bool) *Tester {
+	d, _ := os.Getwd()
+	klog.Info(d)
+
+	var err error
 	if tester == nil {
-		tester = createTester(ctx)
+		tester, err = createTester(ctx, controllerClientsPresence, ignoreClusterIDError)
+		if err != nil {
+			klog.Fatal(err)
+		}
 	}
-
 	return tester
 }
 
-func createTester(ctx context.Context) *Tester {
-	kubeconfig1 := os.Getenv("KUBECONFIG_1")
-	if kubeconfig1 == "" {
-		klog.Error("KUBECONFIG_1 not set")
-		os.Exit(1)
-	}
-	kubeconfig2 := os.Getenv("KUBECONFIG_2")
-	if kubeconfig2 == "" {
-		klog.Error("KUBECONFIG_2 not set")
-		os.Exit(1)
-	}
-	namespace := os.Getenv("NAMESPACE")
-	if namespace == "" {
-		klog.Error("NAMESPACE not set")
-		os.Exit(1)
-	}
+func createTester(ctx context.Context, controllerClientsPresence, ignoreClusterIDError bool) (*Tester, error) {
+	namespace := testutils.GetEnvironmentVariable(namespaceEnvVar)
+	TmpDir := testutils.GetEnvironmentVariable(KubeconfigDirVarName)
 
-	config1, err := clientcmd.BuildConfigFromFlags("", kubeconfig1)
-	if err != nil {
-		klog.Error(err)
-		os.Exit(1)
-	}
-	config2, err := clientcmd.BuildConfigFromFlags("", kubeconfig2)
-	if err != nil {
-		klog.Error(err)
-		os.Exit(1)
-	}
-	clientset1, err := kubernetes.NewForConfig(config1)
-	if err != nil {
-		klog.Error(err)
-		os.Exit(1)
+	// Here is necessary to add the controller runtime clients.
+	scheme := getScheme()
+
+	tester = &Tester{
+		Namespace: namespace,
 	}
-	clientset2, err := kubernetes.NewForConfig(config2)
+
+	tester.ClustersClients = map[string]client.Client{}
+	clusterNumber, err := getClusterNumberFromEnv()
 	if err != nil {
-		klog.Error(err)
-		os.Exit(1)
+		return nil, err
 	}
-	clusterID1, err := utils.GetClusterIDWithNativeClient(ctx, clientset1, namespace)
-	if err != nil {
-		klog.Warningf("an error occurred while getting cluster-id configmap %s", err)
-		clusterID1 = ""
+
+	for i := 1; i <= clusterNumber; i++ {
+		var kubeconfigName = fmt.Sprintf("%s_%d", kubeconfigBaseName, i)
+		var kubeconfigPath = filepath.Join(TmpDir, kubeconfigName)
+		if _, err = os.Stat(kubeconfigPath); err != nil {
+			return nil, err
+		}
+		var c = ClusterContext{
+			Config:         testutils.GetRestConfig(kubeconfigPath),
+			KubeconfigPath: kubeconfigPath,
+		}
+		c.NativeClient = testutils.GetNativeClient(c.Config)
+		c.ClusterID, err = testutils.GetClusterID(ctx, c.NativeClient, namespace)
+		if err != nil && !ignoreClusterIDError {
+			return nil, err
+		}
+
+		if controllerClientsPresence {
```

Comment:
Would always initialize the clients, to avoid having undefined variables which might lead to unexpected crashes.